assets/src/models/createDetourMachine.ts

```diff
@@ -1,17 +1,142 @@
-import { setup } from "xstate"
+import { setup, assign, fromPromise } from "xstate"
+import { RouteId, RoutePattern, RoutePatternId } from "../schedule"
+import { fetchRoutePatterns } from "../api"
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.edit.done" } | { type: "detour.edit.resume" }
+    context: {
+      routeId?: RouteId
+      routePatternId?: RoutePatternId
+
+      routePatterns?: RoutePattern[]
+    }
+
+    input:
+      | {
+          // Caller has target route pattern
```

Comment:
note(minor): I'd have put these comments before the blocks, but then the formatter made it look weird